src/ui/Factory.cpp

```diff
@@ -89,8 +89,8 @@ std::shared_ptr<Step> Factory::createStep(const std::string &command) {
     }
 }
 
-std::shared_ptr<Machine> Factory::createMachine(const State &state) {
-    return std::make_shared<Machine>(nullptr, shared_from_this(), state);
+std::shared_ptr<Machine> Factory::createMachine(const State &state) const {
+    return std::make_shared<Machine>(nullptr, Factory::create(reader(), printer()), state);
```

Comment:
Another constructor for `Machine` with no `Model` parameter? nullptr is confusing.

Reply:
I don't understand this change